go/vt/vtctl/reparentutil/replication.go

```diff
@@ -153,6 +153,18 @@ func ReplicaWasRunning(stopStatus *replicationdatapb.StopReplicationStatus) (boo
 	return stopStatus.Before.IoState == int32(mysql.ReplicationStateRunning) || stopStatus.Before.SqlState == int32(mysql.ReplicationStateRunning), nil
 }
 
+// SQLThreadWasRunning returns true if a StopReplicationStatus indicates that the
+// replica had a running sql thread. It returns an
+// error if the Before state of replication is nil.
+func SQLThreadWasRunning(stopStatus *replicationdatapb.StopReplicationStatus) (bool, error) {
+	if stopStatus == nil || stopStatus.Before == nil {
+		return false, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "could not determine Before state of StopReplicationStatus %v", stopStatus)
+	}
+
+	replStatus := mysql.ProtoToReplicationStatus(stopStatus.Before)
+	return replStatus.SQLState == mysql.ReplicationStateRunning, nil
```

Comment:
This will be backwards compatible once https://github.com/vitessio/vitess/pull/10167 is merged.